app/routes/session.py

```diff
@@ -73,6 +73,20 @@ def login() -> Response:
 
     validate_jti(decrypted_token)
 
+    logger_args = {
+        key: value
+        for key, value in {
+            "tx_id": decrypted_token.get("tx_id"),
+            "case_id": decrypted_token.get("case_id"),
+            "schema_name": decrypted_token.get("schema_name"),
+            "schema_url": decrypted_token.get("schema_url"),
+            "ru_ref": decrypted_token.get("ru_ref"),
```

Comment:
in the case of auth payload v2, I I think `ru_ref` will be under `survey_metadata`
so I'm seeing it in debug as `decrypted_token["survey_metadata"]["data"]["ru_ref"]` and maybe same for qid? I think you might need an if else/ternary for that case

Reply:
Updated now

Reply:
the update still isn't quite working as it has changed to `decrypted_token.get("survey_metadata", {}).get("ru_ref")`
but it needs to be `decrypted_token.get("survey_metadata", {}).get("data", {}).get("ru_ref")`